crates/uv-distribution/src/pyproject_mut.rs

```diff
@@ -0,0 +1,156 @@
+use std::fmt;
+use std::str::FromStr;
+
+use thiserror::Error;
+use toml_edit::{Array, DocumentMut, Item, RawString, TomlError, Value};
+
+use pep508_rs::Requirement;
+use pypi_types::{LenientRequirement, VerbatimParsedUrl};
+
+use crate::pyproject::PyProjectToml;
+
+/// Raw and mutable representation of a `pyproject.toml`.
+///
+/// This is useful for operations that require editing an existing `pyproject.toml` while
+/// preserving comments and other structure, such as `uv add` and `uv remove`.
+pub struct PyProjectTomlMut {
+    doc: DocumentMut,
+}
+
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("Failed to parse `pyproject.toml`")]
+    Parse(#[from] Box<TomlError>),
+    #[error("Dependencies in pyproject.toml are malformed")]
```

Comment:
Nit: wrap pyproject.toml in backticks for consistency with the other error variant.